web/global.d.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * @copyright 2016-present Kriasoft (https://git.io/vMINh)
+ */
+
+import {} from "react";
+import { Theme } from "@emotion/react";
+import { Interpolation } from "@emotion/serialize";
+
+declare module "@emotion/react" {
+  interface Theme {
+    mode: "light" | "dark";
+
+    palette: {
+      primary: string;
+      secondary: string;
+    };
+
+    typography: {
+      fontFamily: string;
+    };
+  }
+}
+
+declare module "react" {
+  interface Attributes {
+    css?: Interpolation<Theme>;
```

Comment:
Have you encountered any problem with `"@emotion/react/types/css-prop"` approach?

Reply:
Nope, it works. Just was experimenting with different approaches.